### PR TITLE
Change Windows hostname on each machine

### LINKED ISSUE #467: Change Windows hostname without rebooting
=== Version ===
* v2.1.1

=== New Enhancement ===
* When booting a custom AMI on AWS with Nanocloud, it would be good to change the hostname.
* It can be done with the following script : 

 _/Set allowed ASCII character codes to Uppercase letters (65..90), 
$charcodes = 65..90

// Convert allowed character codes to characters
$allowedChars = $charcodes | ForEach-Object { [char][byte]$_ }

$LengthOfName = 10
// Generate computer name
$pw = ($allowedChars | Get-Random -Count $LengthOfName) -join ""
$prefix = "CDW12"
$name = $prefix+$pw

$ComputerName = $name
   
Remove-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters" -name "Hostname" 
Remove-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters" -name "NV Hostname" 

New-PSDrive -name HKU -PSProvider "Registry" -Root "HKEY_USERS"

Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Control\Computername\Computername" -name "Computername" -value $ComputerName
Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Control\Computername\ActiveComputername" -name "Computername" -value $ComputerName
Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters" -name "Hostname" -value $ComputerName
Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters" -name "NV Hostname" -value  $ComputerName
Set-ItemProperty -path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" -name "AltDefaultDomainName" -value $ComputerName
Set-ItemProperty -path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" -name "DefaultDomainName" -value $ComputerName
#Set-ItemProperty -path "HKU:\.Default\Software\Microsoft\Windows Media\WMSDK\General" -name "Computername" -value $ComputerName_

Add a new entry in HKLM\SOFTWARE\Microsoft\Windows\CurrentVersion\RunOnce

### PR DESCRIPTION
**Pull request by Leblantoine**
Fixes #467 #451 

To have a unique hostname on each machine we have to add a script in the AMI and create a registry key before create an image. This registry key has the effect to run the script on start.

### CODE REVIEW

File: api/services/MachineService.js [hunk 1]
```diff
@@ -826,8 +826,28 @@ function createImage(image) {
...
+          `add`,
+          `HKLM\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\RunOnce`,
+          `/v`,
+          `ChangeComputerName`,
+          `/t`,
+          `REG_SZ`,
+          `/d`,
+          `"C:\\WINDOWS\\system32\\WindowsPowerShell\\v1.0\\powershell.exe -command ' & C:\\Windows\\changeComputerName.ps1'"`,
+          `/f`
+        ],
```
Comment (Gentux) [inline api/services/MachineService.js:842]: What happen if this script is not available in your AMI ?

Comment (Leblantoine) [reply api/services/MachineService.js:842]: I test with qemu, nothing happen.

File: image/changeComputerName.ps1 [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+New-PSDrive -name HKU -PSProvider "Registry" -Root "HKEY_USERS"
+
+Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Control\Computername\Computername" -name "Computername" -value $ComputerName
+Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Control\Computername\ActiveComputername" -name "Computername" -value $ComputerName
+Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters" -name "Hostname" -value $ComputerName
+Set-ItemProperty -path "HKLM:\SYSTEM\CurrentControlSet\Services\Tcpip\Parameters" -name "NV Hostname" -value $ComputerName
+Set-ItemProperty -path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" -name "AltDefaultDomainName" -value $ComputerName
+Set-ItemProperty -path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon" -name "DefaultDomainName" -value $ComputerName
+#Set-ItemProperty -path "HKU:.Default\Software\Microsoft\Windows Media\WMSDK\General" -name "Computername" -value $ComputerName
```
Comment (dynamiccast) [inline image/changeComputerName.ps1]: Why is this line commented ?
Can you provide the origin of the script ?

Comment (Leblantoine) [reply image/changeComputerName.ps1]: The original script is here : #467

Comment (dynamiccast) [reply image/changeComputerName.ps1]: Did @NanoMarc wrote this script from scratch ?
I am looking for the real source here.

Comment (marc-cabocel) [reply image/changeComputerName.ps1]: No no I didn't write that script from scratch. I took it from here : https://gist.github.com/timnew/2373475

**Review by corentindrouet — APPROVED** [review]
This pr works great

**Pull request merged**